core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -173,6 +238,28 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
     }
   }
 
+  @Override public SqlNode getCastCall(
+      SqlNode operandToCast, RelDataType castFrom, RelDataType castTo) {
+    if (castTo.getSqlTypeName() == SqlTypeName.TIMESTAMP && castTo.getPrecision() > 0) {
+      return new CastCallBuilder(this).makCastCallForTimestampWithPrecision(operandToCast,
+          castTo.getPrecision());
+    } else if (castTo.getSqlTypeName() == SqlTypeName.TIME && castTo.getPrecision() > 0) {
+      return makCastCallForTimeWithPrecision(operandToCast, castTo.getPrecision());
+    }
+    return super.getCastCall(operandToCast, castFrom, castTo);
+  }
+
+  private SqlNode makCastCallForTimeWithPrecision(SqlNode operandToCast, int precision) {
+    SqlParserPos pos = SqlParserPos.ZERO;
+    SqlNode timeWithoutPrecision =
+        getCastSpec(new BasicSqlType(RelDataTypeSystem.DEFAULT, SqlTypeName.TIME));
+    SqlCall castedTimeNode = CAST.createCall(pos, operandToCast, timeWithoutPrecision);
+    SqlCharStringLiteral timeFormat = SqlLiteral.createCharString("HH24:MI:SS.S(" + precision + ")",
```

Comment:
Please use String.format for concatenation

Reply:
Done